packages/components/src/List/Toolbar/ColumnChooserButton/README.md

```diff
@@ -0,0 +1,274 @@
+
+The component holds reference :
+* Visibility: [RowCheckbox](#rowcheckbox)
+* Label: [RowLabel](#rowlabel)
+
+| Props | Type | Info
+|---|---|---|
+| className | string | class passed to div wrapper row
+| children | React.element, [React.element] | the row's  content of the popover
+
+### RowCheckbox
+
+Add a checkbox to switch the visibility of a column.
+If the column is locked, a locked icon replaces the checkbox.
+	dataFeature,
+	describedby,
+	description,
+	id,
+	label,
+	locked = false,
+	onClick,
+	checked = false,
+
+| Props | Type | Info
+|---|---|---|
+| dataFeature | string | data-feature attribute of the input
+| describedby | string | the id of the markup holding the description
+| description | string | the description
+| id | string | id of the checkbox input
+| label | string | label used
+| locked | bool | indicates if the column is locked or not
+| onChange | func | trigger when the checkbox is clicked
+| checked | bool | indicates the visibility value, and if the checkbox is checked
+
+### RowLabel
+
+Add the label of the column to the row.
+
+| Props | Type | Info
+|---|---|---|
+| label | string | the label of the column
+
+## Hooks
+### ColumnChooserManagerHook
+
+This hook manages all the logic behavior in the [ColumnChooser](#columnchooser).
+It holds the state of the columns edited by the user.
+
+| Parameters | Type | Info
+|---|---|---|
+| initColumns | array | array of columns object from the list
+| nbLockedLeftItems | number | the number of columns that will be locked, left to right
+
+And it returns an object with hook state values and handlers to manipulate them.
+
+| Fields | Type | Info
+|---|---|---|
+| onChangeVisibility | function | change the visibility of the given column index|
+| onSelectAll | function | change the visibility of every columns
+| columnsChooser | [ColumnsChooserPropTypes](#columnschooserproptypes) | the hook columns
+| selectAll | bool | state of the select all button, checked or not.
+
+## Context
+
+### columnChooserContext
+
+The column chooser uses a context to pass some props to its children and avoid props drilling.
+By default the context is initialized in [ColumnChooser](#columnchooser).
+You can import directly the provider and the consumer from columnChooser.context
+```
+import { columnChooserContext, ColumnChooserProvider } 
+															from '.../columnChooserContext'
+```
+
+## PropTypes and Structures
+
+### ColumnChooserPropTypes
+
+| Props | Type | Info
+|---|---|---|
+| label | PropTypes.string | the label of the column
+| locked | PropTypes.bool | indicates the locked state
+| order | PropTypes.number | indicates the ordering of the column
+| visible | PropTypes.bool | the state of visibility of the column
+
+
+### ColumnsChooserPropTypes
+
+An array of [ColumnChooserPropTypes](#columnchooserproptypes)
+
+## Utils
+### mergedColumnsChooser
+Helper to merge the columns chooser collection into the columns list.
```

Comment:
This is not clear. Can you describe its api ?

Reply:
What about : The columns you will receive from the column chooser are not synced with the one from the list.
So you will have to use this helper to merge the content given by column chooser with the one from the list.